### ISSUE TITLE
`unneeded_escaping` False Negative in Version 0.63.0-rc.2

### ISSUE THREAD

**Issue by F1248** [realm/SwiftLint#6385]
### New Issue Checklist

- [X] I've Updated SwiftLint to the latest version.
- [X] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

As simplified as possible:

```swift
func function(closure: @escaping () -> Void) { // Should trigger `unneeded_escaping`
	_ = closure
}
```

Command:

```zsh
swiftlint lint --only-rule unneeded_escaping --no-cache File.swift
```

### Environment

* SwiftLint version: 0.63.0-rc.2
